app/services/notification_subscriptions/toggle.rb

```diff
@@ -0,0 +1,94 @@
+module NotificationSubscriptions
+  class Toggle
+    attr_reader :current_user, :permitted_params
+
+    def self.call(...)
+      new(...).call
+    end
+
+    def initialize(current_user, permitted_params)
+      @current_user = current_user
+      @permitted_params = permitted_params
+    end
+
+    def call
+      toggle_subscription
+    end
+
+    def toggle_subscription
+      action = permitted_params[:action]
+
+      if action == "unsubscribe"
+        subscription_id = permitted_params[:subscription_id]
+        notification = NotificationSubscription&.find(subscription_id) unless subscription_id.nil?
```

Comment:
I guess I still find the `&.find` syntax here odd — and it still seems like this should be scoped to `current_user`, where the `&.find` would make a little more sense. Can you help me understand what's happening here?